### PR TITLE
Get Proposal rest endpoints for both Requestor and Provider

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
resolves: #385 
resolves: #384 

It would be good to write rest endpoint test, but I would do it after merging:
https://github.com/golemfactory/yagna/pull/438
where some useful test utils are present.

### CODE REVIEW

File: core/market/decentralized/src/market.rs [hunk 1]
```diff
@@ -158,6 +160,22 @@ impl MarketService {
         // TODO: shouldn't remove precede negotiation unsubscribe?
         Ok(self.matcher.unsubscribe_demand(demand_id, id).await?)
     }
+
+    pub async fn get_proposal(
+        &self,
+        proposal_id: &ProposalId,
+        _id: &Identity,
+    ) -> Result<Proposal, MarketError> {
+        // TODO: Autorization
```
Comment (jiivan) [inline core/market/decentralized/src/market.rs:169]: What are the dangers of unauthorized access to proposal?

**Review by jiivan — APPROVED** [review]
lgtm

File: core/market/decentralized/src/market.rs (hunk 1 above)
Comment (nieznanysprawiciel) [reply core/market/decentralized/src/market.rs:169]: Negotiations are private, meaning that only negotiating parties should know content of Proposals (Other identities on the same daemon probably shouldn't have access either).
But we scheduled thinking about Identities to Milestone 2, so we may decide, that we don't need authorization here.

File: core/market/decentralized/tests/test_proposal.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+use ya_market_decentralized::testing::proposal_util::exchange_draft_proposals;
+use ya_market_decentralized::testing::{GetProposalError, MarketsNetwork, OwnerType};
+
+use ya_client::model::market::proposal::State;
+
+#[cfg_attr(not(feature = "market-test-suite"), ignore)]
+#[actix_rt::test]
+async fn test_get_proposal_rest() -> Result<(), anyhow::Error> {
```
Comment (tworec) [inline core/market/decentralized/tests/test_proposal.rs:8]: REST test is missing.

**Review by tworec — CHANGES_REQUESTED** [review]

File: core/market/decentralized/tests/test_proposal.rs (hunk 2 above)
Comment (jiivan) [reply core/market/decentralized/tests/test_proposal.rs:8]: added

File: core/market/decentralized/tests/test_proposal.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    assert_eq!(proposal.issuer_id()?, &prov_id.identity.to_string());
+    assert!(proposal.prev_proposal_id().is_ok());
+    Ok(())
+}
+
+/// Try to query not existing Proposal.
+#[cfg_attr(not(feature = "market-test-suite"), ignore)]
+#[actix_rt::test]
+async fn test_get_proposal_not_found() -> Result<(), anyhow::Error> {
+    let network = MarketsNetwork::new("test_get_proposal_rest")
```
Comment (tworec) [inline core/market/decentralized/tests/test_proposal.rs:51]:
```suggestion
    let network = MarketsNetwork::new("test_get_proposal_not_found")
```

Comment (tworec) [reply core/market/decentralized/tests/test_proposal.rs:51]: Will change it within conflicts resolve

**Review by tworec — CHANGES_REQUESTED** [review]
My review in form of PR #579

**Review by tworec — APPROVED** [review]

**Pull request merged**